### ISSUE TITLE
[req] Use global/custom proxy

### ISSUE THREAD

**Issue by qg0** [Jackett/Jackett#8503]
### Environment

**OS**: SUSE Linux
**Jackett Version**: 0.16.256.0

**Are you using a proxy or VPN?** [yes/no]
YES. TOR SOCKS5 proxy.

### Description
Adding the tracker https://www.cinecalidad.is/espana/

### Logged Error Messages
```
2020-05-03 08:01:39 | Error | System.Exception: Request to https://www.cinecalidad.is/espana/ failed (Error Forbidden) - The page is protected by an Cloudflare reCaptcha. The page is in aggressive DDoS mitigation mode or your IP might be blacklisted (e.g. in case of shared VPN IPs). There's no easy way of making it usable with Jackett.    at Jackett.Common.Indexers.BaseWebIndexer.RequestStringWithCookiesAndRetry(String url, String cookieOverride, String referer, Dictionary`2 headers) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/BaseIndexer.cs:line 458    at Jackett.Common.Indexers.Cinecalidad.PerformQuery(TorznabQuery query) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/Cinecalidad.cs:line 100    at Jackett.Common.Indexers.Cinecalidad.ApplyConfiguration(JToken configJson) in /home/vsts/work/1/s/src/Jackett.Common/Indexers/Cinecalidad.cs:line 71    at Jackett.Server.Controllers.IndexerApiController.UpdateConfig(ConfigItem[] config) in /home/vsts/work/1/s/src/Jackett.Server/Controllers/IndexerApiController.cs:line 94    at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited\|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited\|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)    at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited\|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited\|19_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)    at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited\|17_0(ResourceInvoker invoker, Task task, IDisposable scope)    at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask\|6_0(Endpoint endpoint, Task requestTask, ILogger logger)    at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)    at Jackett.Server.Middleware.CustomExceptionHandler.Invoke(HttpContext httpContext) in /home/vsts/work/1/s/src/Jackett.Server/Middleware/CustomExceptionHandler.cs:line 70
-- | -- | --
2020-05-03 08:01:38 | Error | On attempt 3 checking for results from Cinecalidad: Request to https://www.cinecalidad.is/espana/ failed (Error Forbidden) - The page is protected by an Cloudflare reCaptcha. The page is in aggressive DDoS mitigation mode or your IP might be blacklisted (e.g. in case of shared VPN IPs). There's no easy way of making it usable with Jackett.
2020-05-03 08:01:36 | Error | On attempt 2 checking for results from Cinecalidad: Request to https://www.cinecalidad.is/espana/ failed (Error Forbidden) - The page is protected by an Cloudflare reCaptcha. The page is in aggressive DDoS mitigation mode or your IP might be blacklisted (e.g. in case of shared VPN IPs). There's no easy way of making it usable with Jackett.
2020-05-03 08:01:33 | Error | On attempt 1 checking for results from Cinecalidad: Request to https://www.cinecalidad.is/espana/ failed (Error Forbidden) - The page is protected by an Cloudflare reCaptcha. The page is in aggressive DDoS mitigation mode or your IP might be blacklisted (e.g. in case of shared VPN IPs). There's no easy way of making it usable with Jackett.
```
***Suggestion***
Some trackers could work only through proxy because the local ISPs blocked local trackers.
Some of them could work without proxy like this http://cinecalidad.is but this site uses Cloudflare reCaptcha when I use proxy like TOR or VPN with shared IPs. So, the situation is like this:

**Without proxy:**
Indexer Test
Tracker1 **UP**
Tracker2 **DOWN** (blocked by local ISP)

**With proxy**:
Indexer Test
Tracker1 **DOWN** (blocked by Cloudflare reCaptcha)
Tracker2 **UP**

Maybe it's a good thing to make an option with a proxy for each tracker? Like a checkbox "use global proxy" and/or text fields "use custom proxy"?

![image](https://user-images.githubusercontent.com/919099/80896435-23592e00-8d18-11ea-9f56-93d232ae016c.png)

**Comment by cadatoiva** [comment]
Proxy use is currently considered experimental in Jackett and is expected to have bugs in it. Using an external VPN to route Jackett traffic instead of the built in proxy settings tends to fix lingering issues like what you describe above.

Until we get the global proxy working correctly, this request will be on hold. But it is a good idea, so thanks for the feedback.

**Comment by ngosang** [comment]
I'm closing because we have more related issues: #7168 #2539
